atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/BackgroundSweeperFastTest.java

```diff
@@ -161,6 +161,46 @@ public void testWriteProgressAfterIncompleteRunWithPreviousProgress() {
                         .build()));
     }
 
+    @Test
+    public void testNewThreadSweepsNewTable() {
+        int otherThreadIndex = THREAD_INDEX + 1;
+        setProgress(ImmutableSweepProgress.builder()
+                .tableRef(TABLE_REF)
+                .staleValuesDeleted(3)
+                .cellTsPairsExamined(11)
+                .minimumSweptTimestamp(4567L)
+                .startRow(new byte[] {1, 2, 3})
+                .startColumn(PtBytes.toBytes("unused"))
+                .timeInMillis(10L)
+                .startTimeInMillis(20L)
+                .build());
+        setNoProgress(otherThreadIndex);
+        setupTaskRunner(OTHER_TABLE, ImmutableSweepResults.builder()
+                .staleValuesDeleted(2)
+                .cellTsPairsExamined(10)
+                .minSweptTimestamp(12345L)
+                .nextStartRow(Optional.of(new byte[] {4, 5, 6}))
+                .timeInMillis(20L)
+                .timeSweepStarted(50L)
+                .build());
+        setNextTableToSweep(OTHER_TABLE);
```

Comment:
A little unclear what code this test is covering, since the cross thread lock contention seems to be mocked and skipped over here.

Reply:
My intention was to test that if one thread is sweeping `TABLE`, then the other thread will sweep some `OTHER_TABLE`.